src/sentry/integrations/github/tasks/link_all_repos.py

```diff
@@ -35,52 +40,64 @@ def link_all_repos(
     integration_id: int,
     organization_id: int,
 ):
-    integration = integration_service.get_integration(
-        integration_id=integration_id, status=ObjectStatus.ACTIVE
-    )
-    if not integration:
-        logger.error(
-            "%s.link_all_repos.integration_missing",
-            integration_key,
-            extra={"organization_id": organization_id},
-        )
-        metrics.incr("github.link_all_repos.error", tags={"type": "missing_integration"})
-        return
 
-    rpc_org = organization_service.get(id=organization_id)
-    if rpc_org is None:
-        logger.error(
-            "%s.link_all_repos.organization_missing",
-            integration_key,
-            extra={"organization_id": organization_id},
-        )
-        metrics.incr(
-            f"{integration_key}.link_all_repos.error",
-            tags={"type": "missing_organization"},
+    with SCMIntegrationInteractionEvent(
+        interaction_type=SCMIntegrationInteractionType.LINK_ALL_REPOS,
+        provider_key=integration_key,
+        organization=organization_id,
+    ).capture() as lifecycle:
+        integration = integration_service.get_integration(
+            integration_id=integration_id, status=ObjectStatus.ACTIVE
         )
-        return
-
-    installation = integration.get_installation(organization_id=organization_id)
-
-    client = installation.get_client()
+        if not integration:
+            # TODO: Remove this logger in favor of context manager
+            logger.error(
+                "%s.link_all_repos.integration_missing",
+                integration_key,
+                extra={"organization_id": organization_id},
+            )
+            metrics.incr("github.link_all_repos.error", tags={"type": "missing_integration"})
+            lifecycle.record_halt(str(LinkAllReposHaltReason.MISSING_INTEGRATION))
+            return
 
-    try:
-        repositories = client.get_repositories(fetch_max_pages=True)
-    except ApiError as e:
-        if installation.is_rate_limited_error(e):
+        rpc_org = organization_service.get(id=organization_id)
+        if rpc_org is None:
+            logger.error(
+                "%s.link_all_repos.organization_missing",
+                integration_key,
+                extra={"organization_id": organization_id},
+            )
+            metrics.incr(
+                f"{integration_key}.link_all_repos.error",
+                tags={"type": "missing_organization"},
+            )
+            lifecycle.record_halt(str(LinkAllReposHaltReason.MISSING_ORGANIZATION))
```

Comment:
Should this be a halt or a failure? Do we expect to encounter this regularly?

Reply:
honestly i have no clue. let me change this to a failure, we can update to halt once we get some more context?